src/Networking/NexusMods.Networking.Downloaders/Tasks/NxmDownloadTask.cs

```diff
@@ -30,15 +31,15 @@ public class NxmDownloadTask : IDownloadTask, IHaveDownloadVersion, IHaveFileSiz
     private long _defaultDownloadedSize;
 
     /// <inheritdoc />
-    public long DownloadedSizeBytes => _state.Job != null ? (long)_state.Job.Current.Value : _defaultDownloadedSize;
+    public long DownloadedSizeBytes => (long)(_state.ActivityStatus?.MakeTypedReport().Current.Value ?? (ulong)_defaultDownloadedSize);
 
     /// <inheritdoc />
-    public long CalculateThroughput<TDateTimeProvider>(TDateTimeProvider provider) where TDateTimeProvider : IDateTimeProvider
+    public long CalculateThroughput()
     {
-        if (_state.Job == null)
+        if (_state.Activity == null)
             return 0;
 
-        return (long)_state.Job.GetThroughput(DateTimeProvider.Instance).Value;
+        return (long)(((ActivityReport<Size>?)_state.ActivityStatus?.GetReport())?.Throughput ?? Size.Zero).Value;
     }
```

Comment:
`ActivityReport<T>.Throughput` appears to never be set by anything, so this will always return `0`.